src/api/shared/create-redis.ts

```diff
@@ -1,5 +1,8 @@
-import IORedis from "ioredis";
+import IORedis, { RedisOptions } from "ioredis";
 import IORedisMock from "ioredis-mock";
+import fs from 'fs';
+
+const fileExists = (path: fs.PathLike) => fs.promises.stat(path).then(() => true, () => false)
 
 export function createRedisFactory(redisUrl?: string): () => IORedis.Redis {
```

Comment:
I think we should maybe add another parameter here and then pass in the environment variable from wherever it's called (only a couple of places). that way, starting the Quirrel server programmatically will still work fine.

Reply:
Sure, could do that. But: wouldn't it still work fine right now? if the REDIS_TLS_CA_FILE is not set, there should not be much of a change compared to the status quo, no? Will add an option to pass in the redis options anyway.

Reply:
That's right, absolutely! It's just an architectural separation I want to maintain, any `process.env` access should only happen in the entrypoints, and everything else should be passed down as config properties. This, for example, allows running a Quirrel server inside the same process as a web server:

```ts
import { runQuirrel } from "quirrel"

runQuirrel({ ..., redisTLS: ... })
```

Reply:
I think I get it, will do.